File: Python/Algorithms/Binary-Search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if target >max(nums) or target<min(nums) or target not in nums:
            return -1
        low = 0
        high = len(nums)
        while low<=high:
            mid = (low+high)//2
            if nums[mid]==target:
                return mid
            elif nums[mid]>=target:
                high = mid-1
            else:
                low = mid+1
        return -1        
            

File: C++/Algorithms/Binary-Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size()-1;
        
        while(low<=high) {
            
            int mid = (low+high)/2;
            
            int item = nums[mid];
            
            if (item == target) {
                return mid;
            }
            
            if (item < target) {
                low = mid+1;
            }
            
            if (item > target) {
                high = mid-1;
            }
        }
        
        return -1;
    }
};